src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -801,9 +1111,8 @@ private void doEncodeOpen(
 
             if (hasMaxFrameSize)
             {
-                amqpOpenRW.maxFrameSize(defaultMaxFrameSize);
+                amqpOpenRW.maxFrameSize(decodeMaxFrameSize);
```

Comment:
The condition on whether or not to set `amqpOpenRW.maxFrameSize(...)` should depend only on the value. If it is not matching the default according to spec, then it should be included, otherwise omitted for brevity.